main/src/ca/uwaterloo/flix/language/phase/jvm/GenAnonymousClasses.scala

```diff
@@ -66,12 +67,21 @@ object GenAnonymousClasses {
   /**
     * Constructor of the class
     */
-  private def compileConstructor(superClass: JvmName, visitor: ClassWriter): Unit = {
+  private def compileConstructor(currentClass: JvmType.Reference, superClass: JvmName, methods: List[JvmMethod], visitor: ClassWriter)(implicit root: Root, flix: Flix): Unit = {
     val constructor = visitor.visitMethod(ACC_PUBLIC, JvmName.ConstructorMethod, MethodDescriptor.NothingToVoid.toDescriptor, null, null)
 
+    // Invoke the superclass constructor
     constructor.visitVarInsn(ALOAD, 0)
     constructor.visitMethodInsn(INVOKESPECIAL, superClass.toInternalName, JvmName.ConstructorMethod,
       MethodDescriptor.NothingToVoid.toDescriptor, false)
+
+    // For each method, compile the closure which implements the body of that method and store it in a field
+    methods.zipWithIndex.foreach { case (m, i) => 
+      constructor.visitVarInsn(ALOAD, 0)
+      GenExpression.compileExpression(m.clo, constructor, currentClass, Map(), new Label())
```

Comment:
I am debating whether or not we should call into GenExpression or rather just copy the 10 or so lines from GenExpression that are relevant. I leaning towards the latter (code duplication, but at the benefit of ensuring there is no dependency between these two parts).

Reply:
Yes, I went through the same thought process. There are several points within `GenExpression` with similar code. There may be some value in bringing them together, but probably not as part of this PR.